controller/scheduler/scheduler.go

```diff
@@ -931,6 +953,11 @@ func (s *Scheduler) handleFormation(ef *ct.ExpandedFormation) (formation *Format
 	log := logger.New("fn", "handleFormation", "app.id", ef.App.ID, "release.id", ef.Release.ID)
 
 	defer func() {
+		// ensure the formation has the correct omni job counts
+		if changed := formation.RectifyOmni(len(s.hosts)); changed {
```

Comment:
Although not strictly needed, I added it as a kind of documentation that the return value indicates that the formation changed, but I think that is clear from the comment of the method, so I've removed it.